driver-core/src/main/java/com/datastax/driver/core/Fetchable.java

```diff
@@ -0,0 +1,166 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import com.google.common.util.concurrent.ListenableFuture;
+
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Defines a result set whose rows can be fetched and paged,
+ * possibly asynchronously.
+ */
+public interface Fetchable<S extends Fetchable<S, T>, T> extends Iterable<T> {
```

Comment:
I like it, it stresses the fact that it extends `Iterable`. `Pageable` sounds nice too; the ticket reporter proposed initially `AsyncPageable`. The votes are open :)